dependencies/direct/py.in

```diff
@@ -1,8 +1,15 @@
 -c py-constraints.in  # limits known broken versions
 
+azure-identity  # credentials.azure_kv
```

Comment:
Are these imported only in the test files directly? If they are runtime deps, move them to the packaging metadata in `pyproject.toml` (`[project].dependencies =`). If some plugins can be optional, their dependencies should probably go into `[project].optional-dependencies =` instead.

Feel free to move the deps updates into a separate PR, though.

Reply:
If I put them in `pyproject.toml` do they ALSO need to be in `py.in` ?

Reply:
`tox` already pulls them in via `package = editable`. And the lock file updater pull in stuff declared in `pyproject.toml`. So everything should be covered by that already.